module/os/linux/spl/spl-kmem-cache.c

```diff
@@ -189,7 +207,31 @@ kv_free(spl_kmem_cache_t *skc, void *ptr, int size)
 	if (current->reclaim_state)
 		current->reclaim_state->reclaimed_slab += size >> PAGE_SHIFT;
 
-	vfree(ptr);
+	kvalloc_endmarker_t *mkr = (kvalloc_endmarker_t *)
+	    ((char *)ptr + size);
+	if (likely(mkr->magic == kvmagic_alive)) {
+		mkr->magic = kvmagic_poison;
+		char *real_allocation_ptr = ptr +
+		    size - mkr->real_allocation_size;
```

Comment:
If my understanding of `real_allocation_size` is correct (that it's the requested size), then it would be the same as `size` here, so `real_allocation_ptr` would be the same as `ptr`.

Reply:
(needs better naming for real_allocation_size - covered by below conversation)